packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart

```diff
@@ -205,6 +205,39 @@ flutter_tools:lib/''');
     KernelCompilerFactory: () => MockKernelCompilerFactory(),
   }));
 
+  test('kernel_snapshot forces platform linking on debug for darwin target platforms', () => testbed.run(() async {
+    final MockKernelCompiler mockKernelCompiler = MockKernelCompiler();
+    when(kernelCompilerFactory.create(any)).thenAnswer((Invocation _) async {
+      return mockKernelCompiler;
+    });
+    when(mockKernelCompiler.compile(
+      sdkRoot: anyNamed('sdkRoot'),
+      aot: anyNamed('aot'),
+      buildMode: anyNamed('buildMode'),
+      trackWidgetCreation: false,
+      targetModel: anyNamed('targetModel'),
+      outputFilePath: anyNamed('outputFilePath'),
+      depFilePath: anyNamed('depFilePath'),
+      packagesPath: anyNamed('packagesPath'),
+      mainPath: anyNamed('mainPath'),
+      extraFrontEndOptions: anyNamed('extraFrontEndOptions'),
+      fileSystemRoots: anyNamed('fileSystemRoots'),
+      fileSystemScheme: anyNamed('fileSystemScheme'),
+      linkPlatformKernelIn: true,
+      dartDefines: anyNamed('dartDefines'),
+    )).thenAnswer((Invocation _) async {
+      return const CompilerOutput('example', 0, <Uri>[]);
+    });
+
+    await const KernelSnapshot().build(androidEnvironment
+      ..defines[kTargetPlatform]  = 'darwin-x64'
+      ..defines[kBuildMode] = 'debug'
+    );
```

Comment:
This test doesn't look like it's testing anything.

Reply:
Hidden on line 226